internal/k8s/controllers/node_controller.go

```diff
@@ -82,6 +82,10 @@ func (r *NodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				level.Error(r.Logger).Log("controller", "NodeReconciler", "error", "old object is not node", "name", oldNodeObj.GetName())
 				return true
 			}
+			if oldNodeObj.Spec.Unschedulable != newNodeObj.Spec.Unschedulable {
```

Comment:
We never return false if they are equal. This must be appended to the below if (or we must flip it to a sequence and change the default, but in that case I'd advise to unit test this)

Reply:
I refactored that part as in this https://github.com/metallb/metallb/pull/2470/commits/64be8c947e8b1051b9d236de8c489ebf9d0d6113

The implementation is a bit more modular and it makes it clear if we want to extend it


Added unit-test also, is there any test case I miss?